Comp2022/src/main/cpp/subsystems/Vision.cpp

```diff
@@ -28,12 +31,33 @@ Vision::Vision()
     // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
 
     table = nt::NetworkTableInstance::GetDefault().GetTable("limelight");
-    table->PutNumber("ledMode", LED_CUR_MODE);
+    table->PutNumber("ledMode", LED_ON);
     table->PutNumber("stream", PIP_SECONDARY);
 
     Initialize();
 }
 
+double Vision::CalculateDist()
+{
+    frc2135::RobotConfig *config = frc2135::RobotConfig::GetInstance();
+    config->GetValueAsDouble("DTL_Distance1", m_distance1, 0.0);
+    config->GetValueAsDouble("DTL_Distance2", m_distance2, 0.0);
+    config->GetValueAsDouble("DTL_VertOffset1", m_vertOffset1, 0.0);
+    config->GetValueAsDouble("DTL_VertOffset2", m_vertOffset2, 0.0);
+
+    m_distance1 = frc::SmartDashboard::GetNumber("DTL_Distance1", m_distance1);
+    m_distance2 = frc::SmartDashboard::GetNumber("DTL_Distance2", m_distance2);
+    m_vertOffset1 = frc::SmartDashboard::GetNumber("DTL_VertOffset1", m_vertOffset1);
+    m_vertOffset2 = frc::SmartDashboard::GetNumber("DTL_VertOffset2", m_vertOffset2);
+    m_slope = (m_distance2 - m_distance1) / (m_vertOffset2 - m_vertOffset1);
+    m_distOffset = m_distance1 - m_slope * m_vertOffset1;
+    frc::SmartDashboard::PutNumber("DTL_Slope", m_slope);
+    frc::SmartDashboard::PutNumber("DTL_Offset", m_distOffset);
+    frc::SmartDashboard::PutNumber("DTL_DistanceLimeLight", m_distOffset);

```

Comment:
For the next few days, I think it would help if we call CalculateDist in Vision::Periodic.  That way, we get these nice helpful SmartDashboard updates every cycle for free. We can look at them when debugging

Reply:
You finished this :+1: